### DISCUSSION TITLE
Arabic not rendered properly (shaped in reverse)

### DISCUSSION THREAD

**Discussion by mustafa0x** [ghostty-org/ghostty#3669] [Feature Requests, Ideas] [answered]
should look like: تجربة

instead, it swaps things around, and looks like this: ةبرجت

<img width="248" alt="image" src="https://github.com/user-attachments/assets/cd3606cf-828b-4a9f-991a-61ad656f031b" />

**Comment by 00-kat** [comment]
I don't think Ghostty currently supports bi-directional text...

**Reply by mustafa0x** [reply]
this isn't bidi text. and if i had to venture a guess, the cause is a harfbuzz misconfiguration. it's working great, just backwards.

**Reply by 00-kat** [reply]
Oh okay, apologies.  I had assumed it was bidi because it was backward.

**Reply by mustafa0x** [reply]
np. bidi is when ltr and rtl are both used in a single "line", which isn't the case here. bidi adds all sort of complexities.

**Comment by mitchellh** [comment, marked as answer]
We force LTR. Bug to fix this in the future: #1442
